app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/AccountBackupDialog.kt

```diff
@@ -121,3 +121,77 @@ private fun NSecCopyButton(
         Text("Copy Secret Key", color = MaterialTheme.colors.onPrimary)
     }
 }
+
+fun Context.getFragmentActivity(): FragmentActivity? {
+    var currentContext = this
+    while (currentContext is ContextWrapper) {
+        if (currentContext is FragmentActivity) {
+            return currentContext
+        }
+        currentContext = currentContext.baseContext
+    }
+    return null
+}
+
+private fun authenticatedCopyNSec(
+    context: Context,
+    scope: CoroutineScope,
+    account: Account,
+    clipboardManager: ClipboardManager,
+) {
+    val fragmentContext = context.getFragmentActivity()!!
+    val authenticators = BIOMETRIC_STRONG or DEVICE_CREDENTIAL

```

Comment:
`BIOMETRIC_STRONG` = fingerprint
`DEVICE_CREDENTIAL` = PIN or password

There is also `BIOMETRIC_WEAK` for face recognition, but I think the best practice would be to use only strong authentication for the nsec whenever possible.